examples/tracing-compare.cpp

```diff
@@ -0,0 +1,178 @@
+                                               BackendKind backendKind) {
+  auto *backend = createBackend(backendKind);
+  Function *F = module.getFunction("resnet50");
+
+  llvm::outs() << "Starting compile.\n";
+  backend->optimizeFunction(CompilationMode::Infer, F);
+  return backend->instrumentAndCompile(F);
+}
+
+std::future<ResultCode> addToDevice(unsigned int id, DeviceManager *device,
+                                    Module &module, FunctionMapTy functions) {
+  std::shared_ptr<std::promise<ResultCode>> compilePromise(
+      new std::promise<ResultCode>);
+  auto future = compilePromise->get_future();
+
+  device->addNetwork(&module, functions,
+                     [compilePromise, id](const Module *, ResultCode code) {
+                       if (code != ResultCode::Ready) {
+                         llvm::errs() << "Failed to compile model for device "
+                                      << id << ".\n";
+                       } else {
+                         llvm::outs()
+                             << "Successfully added to Device " << id << ".\n";
+                       }
+                       compilePromise->set_value(code);
+                     });
+
+  return future;
+}
+
+int main(int argc, char **argv) {
+  llvm::cl::ParseCommandLineOptions(
+      argc, argv, "Run resnet and export a json file containing trace events");
+
+  std::array<DeviceManager *, supportedBackends.size()> devices;
+  for (unsigned int i = 0; i < supportedBackends.size(); ++i) {
+    devices[i] = DeviceManager::createDeviceManager(supportedBackends[i],
+                                                    "tracing-compare");
+    devices[i]->init();
+  }
+
+  // Load and compile model.
+
+  Module module;
+  TypeRef inputType(module.uniqueType(ElemKind::FloatTy, {1, 3, 224, 224}));
+  Placeholder *input, *output;
+
+  std::tie(input, output) = loadResnet50Model(inputType, module);
+
+  std::array<std::unique_ptr<CompiledFunction>, supportedBackends.size()>
+      compiledFunctions;
+
+  for (unsigned int i = 0; i < supportedBackends.size(); ++i) {
+    compiledFunctions[i] = compileModel(module, supportedBackends[i]);
+
+    FunctionMapTy functions;
+    functions.emplace("resnet50", compiledFunctions[i].get());
+
+    auto f = addToDevice(i, devices[i], module, functions);
+    f.wait_for(/* timeout_duration */ std::chrono::seconds(30));
+    if (f.get() != ResultCode::Ready) {
+      return 1;
+    }
+  }
+
+  auto image =
+      readPngImageAndPreprocess(inputImage, ImageNormalizationMode::k0to1,
+                                ImageChannelOrder::BGR, ImageLayout::NCHW,
+                                /* useImagenetNormalization */ true);
+
+  Tensor batch = image.getUnowned(inputType->dims());
+
+  llvm::outs() << "Starting Run.\n";
+  std::array<std::promise<std::unique_ptr<Context>>, supportedBackends.size()>
+      promises;
+
+  for (unsigned int i = 0; i < supportedBackends.size(); ++i) {
```

Comment:
Minor stylistic nit: 
We usually use a `for`-style like this: `for (unsigned i = 0, e = supportedBackends.size(); i < e; ++i){}`